### DISCUSSION TITLE
How does the audio transmisison work?

### DISCUSSION THREAD

**Discussion by 404cofee** [qwbarch/mirage#94] [Q&A] [answered]
I confused myself with the last one let me phrase it in a better way what I wanted to ask is if the audio is shared between everyone for instance audio of you screaming from a session they weren't in and if so can I go through my mirage audio folder to remove or add audio like prerecorded lines and I have another question does the name of the audio matter for it to be registered so I can sort through some of them and name them like "Hello.wav".

Just asking because I do not know how this works and I am just curious about it and also the idea of prerecording lines is really funny to me because if I can record variations of some things I commonly say or add things to trip them up with man that would make things much more interesting.

**Comment by qwbarch** [comment, marked as answer]
The way Mirage works is like this: each player only saves their own audio. When a monster spawns, they pick a player they want to mimic. The player it's mimicking will send that audio to all other players to play on their side (this is how the synced audio works).  
Since each player only holds their own audio, it requires you to be in the game for that player's voice to be mimicked.

Does that make sense? I can try to clarify a bit more if you still have any questions.

**Reply by 404cofee** [reply]
If I edit the audio and delete and add the edited audio does it require to be a certain name or encoded in a certain way?

**Reply by 404cofee** [reply]
I just figured it out had to make them the same names as the files that already existed, With that being said I am to lazy to go to the suggestions and just will ask straight up do you think you could make it so we can name the files and have them work without the folder deleting itself because names look similar and I hate changing meta data for this sort of stuff.

If not no worries because people who want to prerecord a line can just do the process I am doing.

**Reply by qwbarch** [reply]
The file must be a ``.wav``, but the filename doesn't matter at all. The mod simply reads all the files inside the audio directory, and saves that list of files to grab from.  

Regarding the folder deletion, you can see how to disable it [here](https://github.com/qwbarch/mirage/issues/98).

**Reply by 404cofee** [reply]
I had the setting on mate so the problem is likely something to do with the mod also what kind of idiot would not check if something has a config or settings before asking stuff like this?

I am 99% sure the problem lies with if you name the audio files that is not something already registered so for instance mirage auto made a clip with the name 17e41179-08f8-4d0d-81a5-de359932736a.wav if I change the name of the audio to mumblingrandom1.wav on boot the game will delete the folder however if I were to replace the audio with a new one if I named it what the mod registered aka "17e41179-08f8-4d0d-81a5-de359932736a.wav" it doesn't delete on boot heck you can even put meta data on the audio file and it doesn't cause a problem.

**Reply by 404cofee** [reply]
Heres a small recording I made by what I mean

https://github.com/user-attachments/assets/8c1e837f-ed26-4de2-abca-e2d3cb880dd5

**Comment by EC574** [comment]
where can i find features listed ? i cant change found the features listed in setting or lethal config

**Reply by qwbarch** [reply]
> where can i find features listed ? i cant change found the features listed in setting or lethal config

If you're talking about the config, here's how you can access it: https://github.com/qwbarch/mirage/discussions/94#discussioncomment-10990692
